addons/point_of_sale/static/src/app/barcode/barcode_reader_service.js

```diff
@@ -1,4 +1,5 @@
 /** @odoo-module **/
+/* global posmodel */
```

Comment:
you cannot rely on `pos` here. The `barcode_reader` service is already a dependency of the `pos` service.

Reply:
@vlst-odoo, Is it better now or should I override `_barcodeGS1Action` just like in Robin's fix ?

Reply:
what you have now is exactly the same thing as before. You are trying to access `pos`, just in a different way. The barcode service is meant to be generic, so it shouldn't reference pos. This would still be the case even if we didn't have the circular dependency problem. 

Pos loyalty doesn't have anything to do about the barcode scanner. The barcode scanner just tells you what barcodes were scanned, it shouldn't perform side effects. 

Asking the barcode scanner to worry about pos loyalty is a bit like asking your mailman to also pay for the invoices that he delivers to you. In principle you could work out an arrangement with him, but that doesn't really work :)

In short, you are right, you should either override one of the existing listeners ( like `_barcodeGS1Action` ) or maybe it would be even better to directly call `useBarcodeReader` again in `pos_loyalty`

Reply:
Thanks for the review and the clear explanation @vlst-odoo I updated the diff :slightly_smiling_face: